src/discord/staff/events.py

```diff
@@ -109,49 +110,52 @@ async def event_remove(
         )
 
         # Check to make sure event has previously been added
-        event_not_in_list = event_name not in [
-            e["name"] for e in src.discord.globals.EVENT_INFO
-        ]
+        event = next(
+            (e for e in src.discord.globals.EVENT_INFO if e.name == event_name),
+            None,
+        )
 
         # Check to see if role exists on server
         server = self.bot.get_guild(env.server_id)
         potential_role = discord.utils.get(server.roles, name=event_name)
 
-        if event_not_in_list and potential_role:
+        # If staff member has selected to delete role from all users, delete role entirely
+        if potential_role:
+            if delete_role == "yes":
+                server = self.bot.get_guild(env.server_id)
+                await potential_role.delete()
```

Comment:
This is subject to deletion of other roles that are *not* events (e.g. `/event remove event:Administrator delete_role:true` will delete the Administrator role). This is a vulnerability, and while it does not escalate privileges for the Staff/VIP role member, it compromises availability, making it a low-skill DoS for Administrators and the like.

The best course of action is to remove this code path, and return back a message saying for the user to delete the role manually within Discord settings to prevent deletion of higher privilege roles.